### ISSUE TITLE
False positive with Style/RedundantLineContinuation

### ISSUE THREAD

**Issue by koppen** [rubocop/rubocop#11844]
## Expected behavior

Running `rubocop --only Style/RedundantLineContinuation` on the following code:

```ruby
# lib/tasks/sample.rake
desc \
  "Here comes a long multiline description"
```

should result in no offenses detected:

> 1 file inspected, no offenses detected

```
lib/tasks/sample.rake:5:10: C: [Correctable] Style/RedundantLineContinuation: Redundant line continuation.
    desc \ ...
```

and running `rubocop --only Style/RedundantLineContinuation --autocorrect` should not touch the file.

## Actual behavior

Running `rubocop --only Style/RedundantLineContinuation` on the following code:

```ruby
# lib/tasks/sample.rake
desc \
  "Here comes a long multiline description"
```

flags the line continuation as being redundant:

```
Offenses:

lib/tasks/import.rake:5:10: C: [Correctable] Style/RedundantLineContinuation: Redundant line continuation.
    desc \ ...
         ^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

(which it isn't) and running _safe_ autocorrections on it changes it to invalid syntax:

```
    desc 
      "Import EOD files from a directory. All files are considered and "
```

which results in

```
$ rake -T
rake aborted!
ArgumentError: wrong number of arguments (given 0, expected 1)
```

## RuboCop version

```
$ bundle exec rubocop -V
1.50.2 (using Parser 3.2.2.1, rubocop-ast 1.28.1, running on ruby 3.1.2) [arm64-darwin21]
  - rubocop-capybara 2.18.0
  - rubocop-rails 2.19.1
  - rubocop-rspec 2.20.0
```

While this looks a lot like a duplicate of https://github.com/rubocop/rubocop/issues/11783, the fix for that issue is already in 1.50.2 as far as I can tell.

**Comment by koic** [comment]
Hhm, I cannot reproduce this issue with the latest master branch and v1.50.2:

```console
$ cat example.rb
desc \
  "Here comes a long multiline description"

$ bundle exec rubocop --only Style/RedundantLineContinuation example.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
```

**Comment by koppen** [comment]
Hm, hang on, let me see what I am missing in the repro code

**Comment by koppen** [comment]
Ah, it needs to be wrapped in a block as well - in this case a `namespace`. I've updated the issue to reflect this.
